x-pack/plugins/maps/public/connected_components/map_container/map_container.tsx

```diff
@@ -36,7 +36,7 @@ import 'mapbox-gl/dist/mapbox-gl.css';
 const RENDER_COMPLETE_EVENT = 'renderComplete';
 
 interface Props {
-  addFilters: ((filters: Filter[]) => Promise<void>) | null;
+  addFilters: ((filters: Filter[]) => void) | null;
```

Comment:
This signature needs to remain as returning `Promise<void>`. This is used in [MapEmbeddable](https://github.com/elastic/kibana/blob/master/x-pack/plugins/maps/public/embeddable/map_embeddable.tsx#L237) and that implementation requires async tasks

Reply:
@nreese Is it okay if update this with `addFilters: ((filters: Filter[]) => Promise<void> | void) | null;`. The `addFilters` prop used in [MapsAppView](https://github.com/elastic/kibana/blob/4ada8b5bf46e101f459125079846771b4a3ac7fc/x-pack/plugins/maps/public/routing/routes/maps_app/maps_app_view.tsx#L441) does not return anything.

Reply:
I would recommend just making the MapsAppView addFilters to be async to keep the method signature simple

Reply:
Thanks @nreese, I have made the changes as you suggested.